### PR TITLE
Migration to ink2

### PR DESCRIPTION
**Pull request by riusricardo**

### CODE REVIEW

File: 0/building-your-contract.md [hunk 1]
```diff
@@ -111,6 +111,16 @@ In the next section we will configure the Polkadot UI.
...
 
+> ** ABI generation has a temporary [bug](https://github.com/paritytech/ink/issues/222#issuecomment-549829912).
+
+ink! provides a built-in overflow protection enabled on our `Cargo.toml` file. It is __recommended__ to keep it enabled as a security mechanism.
+```
+[profile.release]
+panic = "abort"
+lto = true
+opt-level = "z"
+overflow-checks = true    <-- Overflow Protection
```
Comment (Robbepop) [inline 0/building-your-contract.md]:
```suggestion
overflow-checks = true    <-- Arithmetic overflow protection
```

File: 0/building-your-contract.md [hunk 2]
```diff
@@ -111,6 +111,16 @@ In the next section we will configure the Polkadot UI.
...
 **Learn More**
 
+> ** ABI generation has a temporary [bug](https://github.com/paritytech/ink/issues/222#issuecomment-549829912).
+
+ink! provides a built-in overflow protection enabled on our `Cargo.toml` file. It is __recommended__ to keep it enabled as a security mechanism.
+```
+[profile.release]
+panic = "abort"
+lto = true
+opt-level = "z"
```
Comment (Robbepop) [inline 0/building-your-contract.md]:
```suggestion
opt-level = "z"            <-- Optimize for small binary output
```

File: 0/building-your-contract.md [hunk 3]
```diff
@@ -111,6 +111,16 @@ In the next section we will configure the Polkadot UI.
 
 **Learn More**
 
+> ** ABI generation has a temporary [bug](https://github.com/paritytech/ink/issues/222#issuecomment-549829912).
+
+ink! provides a built-in overflow protection enabled on our `Cargo.toml` file. It is __recommended__ to keep it enabled as a security mechanism.
+```
+[profile.release]
+panic = "abort"
```
Comment (Robbepop) [inline 0/building-your-contract.md]:
```suggestion
panic = "abort"             <-- Panics shall be treated as aborts: reduces binary size
```

File: 0/building-your-contract.md [hunk 4]
```diff
@@ -111,6 +111,16 @@ In the next section we will configure the Polkadot UI.
...
+lto = true
```
Comment (Robbepop) [inline 0/building-your-contract.md]:
```suggestion
lto = true             <-- enable link-time-optimization: more efficient codegen
```

File: 0/building-your-contract.md [hunk 5]
```diff
@@ -111,6 +111,16 @@ In the next section we will configure the Polkadot UI.
...
 
+> ** ABI generation has a temporary [bug](https://github.com/paritytech/ink/issues/222#issuecomment-549829912).
+
+ink! provides a built-in overflow protection enabled on our `Cargo.toml` file. It is __recommended__ to keep it enabled as a security mechanism.
+```
+[profile.release]
+panic = "abort"             <-- Panics shall be treated as aborts: reduces binary size
+lto = true             <-- enable link-time-optimization: more efficient codegen
+opt-level = "z"            <-- Optimize for small binary output
+overflow-checks = true    <-- Arithmetic overflow protection
```
Comment (Robbepop) [inline 0/building-your-contract.md]: oups, maybe we want to align the arrows :sweat_smile:

**Comment by shawntabrizi** [comment]
Make sure you keep to the ERC20 standard: https://github.com/paritytech/ink/pull/231

File: 0/building-your-contract.md [hunk 6]
```diff
@@ -19,14 +19,14 @@ The final, optimized `flipper.wasm` file is what we will actually deploy to our
 ## Contract ABI
 By running the next command we'll generate the Application Binary Interface (ABI):
```
Comment (Stefie) [inline 0/building-your-contract.md]: See discussion about correct wording here https://github.com/paritytech/ink/pull/236#discussion_r345732928

File: 0/calling-your-contract.md [hunk 7]
```diff
@@ -17,9 +17,9 @@ When you press "Call" you will see it returns the value `false`:
 
 ![An image of Flipper RPC call with false](./assets/flipper-false.png)
 
-You might wonder: "Why did we need to specify gas when reading a value from a contract?"
-
-
+> NOTE: You might be wondering: "Why did we need to specify gas when reading a value from a contract?"
+> 
+> If you notice right above the "Call" button is a toggle which allows you to "send call as transaction" or "send as RPC call". For a read-only request like this, we can simply use an RPC call which will _simulate_ a transaction, but not actually store anything on-chain. Thus, you will still need to specify the right amount of gas to cover your "virtual fee", but don't worry, nothing will be charged when making a call this way. :)
```
Comment (Robbepop) [inline 0/calling-your-contract.md]: A side benefit is that we instantly see the returned value of a call with an RPC call.

**Pull request merged**